research/src/pages/file.proposal.mdx

```diff
@@ -0,0 +1,165 @@
+menu: Components
+name: File (Editor's Draft)
+path: /components/file
+pathToResearch: /components/file.research
+---
+
+# File Input Component Specification
+
+## Overview <a href="#overview" id="overview"></a>
+
+The File Input component is a control that provides the user with the ability
+to select local files.
+
+### Background <a href="#background" id="background"></a>
+
+File Input controls are widely used and implemented on the web. They are as of
+today, the only thing that allows an application to access the local file
+system, in a fairly limited but controlled way.
+
+### Use Cases <a href="#use-cases" id="use-cases"></a>
+
+The `<input type=file>` control is widely used to provide applications with the
+capability of reading local files on form submission, or uploading files to a
+server.
+
+### Features <a href="#features" id="features"></a>
+
+The main feature is selecting local files, potentially receiving hints from the
+website about which file kinds are allowed, for example.
+
+### Prior Art/Examples <a href="#prior-art" id="prior-art"></a>
+
+#### Browser implementations
+
+Browsers show an interoperable rendering for file inputs, consisting of a
+button and a label.
+
+- WebKit and Blink use an `<input type=button>`, and a magical label that
+  doesn't show up in the DOM.
+
+- Gecko uses an actual `<button tabindex=-1>` and an HTML `<label>` element.
+
+Those three implementations use some form of middle-cropping when the selected
+file name is too large.
+
+#### Custom inputs on the web
+
+Custom inputs on the web are generally achieved by hiding the original input.
+Some approach it by hiding it in a way that the user can still hit (like
+`opacity: 0`), others use JS to forward the click from some other UI element to
+the underlying file input.
+
+- [Bootstrap](https://getbootstrap.com/docs/4.0/components/forms/#file-browser)
+  uses the former approach, by generating a "Browse" button with generated
+  content.
+
+- [Carbon Design System](https://www.carbondesignsystem.com/components/file-uploader/code/)
+  uses the later.
+
+---
+
+### API <a href="#api" id="api"></a>
+
+See [the Resources section](#resources) for the specification and the MDN reference.
+
+#### Properties and Attributes <a href="#properties-attributes" id="properties-attributes"></a>
+
+See [the Resources section](#resources) for the specification and the MDN reference.
+
+#### Events <a href="#events" id="events"></a>
+
+See [the Resources section](#resources) for the specification and the MDN reference.
+
+### Appearance <a href="#appearance" id="appearance"></a>
+
+![File Chooser Example](../images/file.png)
+
+### Anatomy <a href="#anatomy" id="anatomy"></a>
+
+TODO
+
+#### Diagram <a href="#diagram" id="diagram"></a>
+
+TODO
+
+#### DOM Structure <a href="#dom-structure" id="dom-structure"></a>
+
+```html
+<button tabindex="-1">Browse...</button>
+<!-- Or appropriately-localized string -->
+<label><!-- implementation and state-dependent --></label>
```

Comment:
The label and button have slots correct? Also based on the name that you have for part these should be referenced with part attributes.

Reply:
I think they should have a _wrapping_ slot, so that you can replace the whole thing, not just the node inside it.